### DISCUSSION TITLE
Create an Integral typeclass?

### DISCUSSION THREAD

**Discussion by bitmappergit** [coalton-lang/coalton#159] [Ideas and Proposals]
Should things like mod be moved to be in an Integral typeclass and div added?

Honestly, a lot of the Integer specific functions might be able to be moved there.

**Comment by stylewarning** [comment]
I've thought the answer should be "yes", but I always wonder when some type classes are too many.

This one is probably entirely reasonable! :)

**Comment by eliaslfox** [comment]
[Here](https://hackage.haskell.org/package/base-4.15.0.0/docs/Prelude.html#t:Integral) is Haskell's Integral type class for reference.
